mcodingbot/bot.py

```diff
@@ -22,15 +26,37 @@ def __init__(self) -> None:
         self.plugins.load_folder("mcodingbot.tasks")
 
         self._session: aiohttp.ClientSession | None = None
+        self._db: Database | None = None
 
     @property
     def session(self) -> aiohttp.ClientSession:
         if not self._session:
             raise AttributeError("Session has not been set yet.")
         return self._session
 
+    @property
+    def db(self) -> Database:
+        if not self._db:
+            raise AttributeError("Database has not been set yet.")
+        return self._db
+
     async def start(self, *args: Any, **kwargs: Any) -> None:
```

Comment:
Is is better to put this stuff in a `create_db` function that's subscribed to `hikari.StartingEvent` imo.

Reply:
Why would we want to create the database after the bot's connected to discord? Events can sometimes trigger before that anyways, it's best to make the DB right away. I could see moving this logic to a method on the Database itself though.

Reply:
`StartingEvent` fires before you connect to discord

Reply:
Okay but subscribing to an event is pointless if you can do it before connecting. It just adds complexity to the bot.

Reply:
I feel like moving the logic to the `Database` class then subscribing the connect function would be the simplest

Reply:
There's no reason to subscribe to an event. Whatever code we call when the StartingEvent is fired can just be put in the start method.

Reply:
I moved the logic to the Database class though.

Reply:
I think that's good enough